blocks/subscriptions-block/components/Offer.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useFusionContext } from 'fusion:context';
+import getProperties from 'fusion:properties';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+
+// TODO: Perhaps this should go in blocks.json as an endpoint entry for the retail API:
+/**
+"api": {
+  "retail": {
+    "origin": "https://corecomponents-the-sun-prod.api.cdn.arcpublishing.com",
+    "endpoint": "/retail/public/v1/offer/live/"
+  }
+}
+ * */
+const OFFER_URL = '/retail/public/v1/offer/live/';
+
+const offerService = ({
+  apiOrigin,
+  campaignCode,
+}) => fetch(`${apiOrigin}${OFFER_URL}${campaignCode || 'default'}`, {})
+  .then((res) => res.json());
+
+const useOffer = ({ campaignCode }) => {
+  const { arcSite } = useFusionContext();
+  const { api: { retail: { origin = null } } } = getProperties(arcSite);
+  const [offer, setOffer] = useState(null);
+  const [error, setError] = useState(null);
+  const [isFetching, setIsFetching] = useState(true);
+
+  const fetchOffer = useCallback(async (code) => {
+    let offerResponse = null;
+    try {
+      offerResponse = await offerService({
+        campaignCode: code,
+        apiOrigin: origin,
+      });
+      setOffer(offerResponse);
+    } catch (err) {
+      setError(`Error in fetching retail offers: ${err.toString()}`);
+    }
+    return offerResponse;
```

Comment:
```suggestion
    try {
      const offerResponse = await offerService({
        campaignCode: code,
        apiOrigin: origin,
      });
      setOffer(offerResponse);
      return offerResponse;
    } catch (err) {
      setError(`Error in fetching retail offers: ${err.toString()}`);
    }
    return null;
```
Minor scope annoyance... (optional, but `let`s feel like code smell to me)

Reply:
Curious! Why do `let`'s feel like code smell to you?

Reply:
Thanks Nick for the suggestion to return null!